kuksa_apps/kanto/edge_client.py

```diff
@@ -0,0 +1,122 @@
+from ditto.protocol.things.commands import Command
+from kuksa_viss_client import KuksaClientThread
+
+from edge_device_info import EdgeDeviceInfo
+from utils import process_tree, process_signal
+
+FEATURE_ID_VSS = "VSS"
+
+EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO = "edge/thing/response"
+EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO_REQUEST = "edge/thing/request"
+
+
+class EdgeClient:
+    def __init__(self):
+        self.mqtt_client = None
+        cfg = {"ip": "127.0.0.1"}
+        self.kuksa_client = KuksaClientThread(cfg)
+        self.device_info = None
+        self.ditto_client = None
+
+    def on_connect(self, client:mqtt.Client, obj, flags, rc):
+        print("Connected with result code " + str(rc))
+        self.mqtt_client = client
+        # init ditto client
+        self.ditto_client = Client(paho_client=self.mqtt_client)
+        self.ditto_client.connect()
+        # init kuksa client
+        self.kuksa_client.start()
+        self.kuksa_client.authorize()
+        # trigger initialization
+        self.mqtt_client.subscribe(EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO)
+        self.mqtt_client.publish(EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO_REQUEST, None, 1)
+
+    def on_message(self, client, userdata, msg):
+        try:
+            if msg.topic == EDGE_CLOUD_CONNECTOR_TOPIC_DEV_INFO:
+                if self.device_info is None:
+                    self.device_info = EdgeDeviceInfo()
+                    self.device_info.unmarshal_json(msg.payload)
+                    self.add_vss_feature()
+                    self.subscribe()
+                else:
+                    print('device info already available - discarding message')
+                return
+        except Exception as ex:
+            print(ex)
+
+    def subscribe(self):
+        self.kuksa_client.subscribe('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Altitude', self.on_kuksa_signal)
+        self.kuksa_client.subscribe('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Latitude', self.on_kuksa_signal)
+        self.kuksa_client.subscribe('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Longitude', self.on_kuksa_signal)
+        # self.kuksa_client.subscribe('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Accuracy', self.on_kuksa_signal)
+        self.kuksa_client.subscribe('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Speed', self.on_kuksa_signal)
+
+    def add_vss_feature(self):
+        # add the vss feature
+        cmd = Command(self.device_info.deviceId).feature(FEATURE_ID_VSS).modify(Feature().to_ditto_dict())
+        cmd_envelope = cmd.envelope(response_required=False, content_type="application/json")
+        self.ditto_client.send(cmd_envelope)
+
+        # add the vss tree as properties
+        vss_tree = self.kuksa_client.getValue('Vehicle/Cabin/Infotainment/Navigation/CurrentLocation/*')
+        processed = process_tree(vss_tree)
+        for key, val in processed.items():
+            if 'Heading' in key or 'Accuracy' in key:
```

Comment:
If changing to `Vehicle.CurrentLocation` (and VSS 3.0) please note that more signals have been added. As an example a signal `Vehicle.CurrentLocation.Timestamp` has been added.

Reply:
If you push a signal that is not in the loaded data model, you will get an error
I think for this PR it is fine, we could make a reminder, to change it after we made 3.0 default, unless @erikbosch  does 2.2 already have Vehicle.CurrentLocation as well? In that case we can change now, and save the trouble later :)

Reply:
`Vehicle.CurrentLocation` was introduced in 2.1 (for 2.1 and 2.2 both paths existed, the old one marked as deprecated)